### PR TITLE
Explain in more details how to build applications using wx

### PR DESCRIPTION
**Pull request by vadz**
We didn't document using wxwin.m4 at all anywhere, so do it at least
minimally now (we still need a separate page documenting its full
functionality) and also show a simple makefile using wx-config.

Also mention different ways of using wx with CMake for completeness.

---

@MaartenBent and/or @TcT2k, could you please check my (untested) instructions for CMake?

Comments about the other changes are welcome too, of course. TIA!

### CODE REVIEW

File: docs/doxygen/overviews/install.md [hunk 1]
```diff
@@ -81,37 +81,132 @@ The wxWidgets ports mentioned above are the main ones, however other variants
...
 
-After installing wxWidgets, it is recommended to check that the library can
-actually be used by building a minimal wxWidgets program using it, such as
-`samples/minimal/minimal.cpp` included in wxWidgets distributions.
+After installing wxWidgets, you need to set up your application to be built
+using it.
 
-You may choose to build this program in any of the following ways:
+As previously, if you're using CMake, please follow the standard way of
+building applications with and use either [add_subdirectory()][1] or
```
[the remark this answers is not in the document]
Comment (vadz) [reply docs/doxygen/overviews/install.md:91]: I had somehow missed this part of our CMake docs, but you're absolutely right, thanks! And yes, Doxygen Markdown can link to the anchors such as `#cmake_apps`.

**Comment by TcT2k** [comment]
> Also mention different ways of using wx with CMake for completeness.
> 
> @MaartenBent and/or @TcT2k, could you please check my (untested) instructions for CMake?
> 

I think the reference to the minimal samples CMakeLists.txt should be sufficient. Strictly speaking we could document that you could either use the classic `find_package(wxWidgets)` but could also use wxwidgets as a subproject via `add_subdirectory()` but I'm not sure if this wouldn't cause more confusion than necessary.

**Comment by PBfordev** [comment]
> ... we could document that you could either use the classic `find_package(wxWidgets)` but could also use wxwidgets as a subproject via `add_subdirectory()` ...

As I wrote, this is already documented: https://docs.wxwidgets.org/trunk/overview_cmake.html#cmake_apps

**Comment by TcT2k** [comment]
Ok sorry, didn't see it there, but looks good and shouldn't cause confusion.

**Pull request closed without merging**